### PR TITLE
Revise blueprint

### PR DESCRIPTION
**Pull request by maasg**
This PR revises the documentation for blueprints to the features included in Cloudflow 2.0

### CODE REVIEW

File: docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
+In Cloudflow, a streaming application is composed of connected `Streamlet`{empty}s.
+Each `Streamlet` defines zero or more inlets and outlets.
+To create a working application, we must define how the `Streamlet`{empty}s connect together, using _topics_ as communication channels.
+ by declaring which outlet from one `Streamlet` should connect with an inlet of another.
 
-A blueprint specifies which streamlets should be used in a pipeline and how they should be connected together.
-It must be defined in the file `src/main/blueprint/blueprint.conf` within the file structure of the Cloudflow application _sbt project_.
+A blueprint specifies the streamlets used in an application and how they connect together.
+In practical terms, a blueprint is a text file that uses the HOCON format to declare the streamlets in an application, the topics used as communication channels, and the connections between streamlets and topics.
+ the It must be defined in the file `src/main/blueprint/blueprint.conf` within the file structure of the Cloudflow application _sbt project_.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc:12]: the It?

Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc:12]:
```suggestion
 The blueprint must be defined in the file `src/main/blueprint/blueprint.conf` within the file structure of the Cloudflow application _sbt project_.
```

File: docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc [hunk 2]
```diff
@@ -18,57 +20,204 @@ include::{cloudflow-examples-version}@docsnippets:ROOT:example$akkastreams-scala
...
+sensor-data = com.example.SensorDataIngress
+----
 
-The streamlet references assigned in the `streamlets` section can be used in the `connections` section to connect streamlets together.
-Lets zoom in on the streamlets section in the example below:
+It is possible to declare additional instances of any streamlet class. 
+This would be a legal declaration:
 
-[source,hocon]
+[source, json]
```
Comment (yuchaoran2011) [inline docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc:81]: Why json? The file content is in hocon.

File: docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc [hunk 3]
```diff
@@ -18,57 +20,204 @@ include::{cloudflow-examples-version}@docsnippets:ROOT:example$akkastreams-scala
...
+In the `topics` section, we define the connections of an application.
+As we explained in xref:app-development-process.adoc[the Cloudflow development process], streamlets have `inlet`{empty}s and `outlet`{empty}s.
+
+=== Consumers and Producers
+
+In the event-based model used by Cloudflow, `inlet`{empty}s are stream consumers and `outlet`{empty}s are stream producers. 
+
+Consumers and producers do not connect point-to-point to each other. 
+Instead, we use topics to provide a communication channel between producers and consumers.
+This model allows for an organic growth of an application over time, as newly implemented processes, modelled as streamlets, can be added to an existing application by connecting them to the list of consumers at the appropriate place in the data flow.
```
Comment (yuchaoran2011) [inline docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc:106]:
```suggestion
This model allows for an organic growth of an application over time, as newly implemented processes, modeled as streamlets, can be added to an existing application by connecting them to the list of consumers at the appropriate place in the data flow.
```

File: docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc [hunk 4]
```diff
@@ -18,57 +20,204 @@ include::{cloudflow-examples-version}@docsnippets:ROOT:example$akkastreams-scala
...
+  producer-config = ${kafka-connection-config} {
+    # more config can be added here from Kafka ProducerConfig (see org.apache.kafka.clients.producer.ProducerConfi)
+  }
+  consumer-config = ${kafka-connection-config} {
+    # more config can be added here from Kafka ConsumerConfig (see org.apache.kafka.clients.consumer.ConsumerConfig)
+  }
+}
+----
+<1> Creates a topic definition by extending the `shared-config` with additional configuration.
+<2> Set the topic name explicitly, using the `topic.name` configuration key.
```
Comment (yuchaoran2011) [inline docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc:175]:
```suggestion
<2> Sets the topic name explicitly, using the `topic.name` configuration key.
```

File: docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc [hunk 5]
```diff
@@ -18,57 +20,204 @@ include::{cloudflow-examples-version}@docsnippets:ROOT:example$akkastreams-scala
...
+
+In the snippet above, we are centralizing the common configuration for all topics in a `shared-config` configuration entry.
+We then use the HOCON configuration extension syntax, as in `${shared-config} {...}` to extend the common configuration with the specific values required for each topic definition.
+
+We are also using the explicit `topic.name` property instead of using the default name of the configuration object.
+
+It is important to note the presence of the flag `managed`, which indicated whether Cloudflow should create the topic (`managed=true` (default)) or not (`managed=false`).
+
+
+We can setup the Kafka provider configuration in a separate configuration file, like this:
```
Comment (yuchaoran2011) [inline docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc:188]:
```suggestion
We can set up the Kafka provider configuration in a separate configuration file, like this:
```

File: docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc (hunk 2 above)
Comment (maasg) [reply docs/shared-content-source/docs/modules/develop/pages/blueprints.adoc]: I was trying to see if the rendering (coloring) was better. Changed back.

**Pull request merged**